File: client/src/app/shared/models/brand.ts
export interface Brand {
  id: string;
  name: string;
  description: string;
  logoUrl: string;
  createdAt: Date;
  updatedAt: Date;
  metadata: Record<string, string>;
  isUsed?: boolean;
}

export interface CreateBrand {
  name: string;
  description: string;
  logoUrl: string;
  metadata?: Record<string, string>;
  isUsed?: boolean;
}

File: Core/Entities/Brand.cs
using System.Collections.Generic;

namespace Core.Entities
{
    public class Brand : BaseEntity
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public string LogoUrl { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        public Dictionary<string, string> Metadata { get; set; } = new Dictionary<string, string>();

        public ICollection<ElectricBikeProduct> ElectricBikeProducts { get; set; }
        public ICollection<AgriculturalMachineProduct> AgriculturalMachineProducts { get; set; }
    }
}

